File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans;
        int n = numRows;
        for (int i = 0; i < n; i++) {
            vector<int> temp(i + 1);
            temp[0] = 1;
            temp[i] = 1;
            if (i >= 2) {
                int j = 1;
                for (int k = 0; k < ans[i - 1].size() - 1; k++) {
                    temp[j] = ans[i - 1][k] + ans[i - 1][k + 1];
                    j++;
                }
            }
            ans.push_back(temp);
        }
        return ans;
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ans = new ArrayList<>();
        for (int i = 0; i < numRows; i++) {
            List<Integer> temp = new ArrayList<>();
            for (int x = 0; x <= i; x++) {
                temp.add(1);
            }
            if (i >= 2) {
                int k = 1;
                for (int j = 0; j < ans.get(i - 1).size() - 1; j++) {
                    temp.set(k,
                    ans.get(i - 1).get(j) +
                    ans.get(i - 1).get(j + 1));
                    k++;
                }
            }
            ans.add(temp);

        }
        return ans;
    }
}